src/components/sections/Newsletter/Newsletter.tsx

```diff
@@ -0,0 +1,82 @@
+import type { ComponentPropsWithRef, FormEvent, ReactNode } from 'react'
+import React, { forwardRef, useRef } from 'react'
+import { Form, Label, Input, Button } from '@faststore/ui'
+
+export interface NewsletterProps
+  extends Omit<ComponentPropsWithRef<'form'>, 'onSubmit'> {
+  /**
+   * Heading for the section.
+   */
+  heading: ReactNode
+  /**
+   * Description about the newsletter.
+   */
+  description?: ReactNode
+  /**
+   * Callback function when submitted.
+   */
+  onSubmit: (value: string) => void
+}
+
+const Newsletter = forwardRef<HTMLFormElement, NewsletterProps>(
+  function Newsletter({ heading, description, onSubmit, ...otherProps }, ref) {
+    const emailInputRef = useRef<HTMLInputElement>(null)
+
+    const handleSubmit = (event: FormEvent) => {
+      event.preventDefault()
+
+      if (emailInputRef.current?.value !== '') {
+        onSubmit(emailInputRef.current!.value)
+      }
+    }
+
+    return (
+      <Form
```

Comment:
Btw, I was wondering if the wrapper should be a `section` and the `data-newsletter-controls` would be the `form`. 🤔  Any thoughts?

Reply:
I added a `section[data-newsletter]` wrapper [here](https://github.com/vtex-sites/base.store/pull/242/commits/3ebbe9edb3eb828f1a8390c706ed441769983f90). However, I kept the form as `form[data-newsletter-form]` instead of `data-newsletter-controls` as controls for me suggests form inputs and not the form itself (_Bootstrap feelings_, it's how they used to name them).